cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DocumentStore.java

```diff
@@ -170,9 +167,15 @@ public void delete() throws IOException {
         if (!location.exists()) {
             String msg = String.format("Datastore %s doesn't exist on disk", location);
             logger.warning(msg);
-            throw new IOException(msg);
+            throw new DatastoreNotDeletedException(msg);
```

Comment:
This is to be consistent with `DatastoreNotOpenedException`, but I agree you have a point on the checked/unchecked argument.